packages/core/src/compose-tunnel-agent-client.ts

```diff
@@ -175,7 +178,12 @@ export const queryTunnels = async ({
       .filter(({ service }: Tunnel) => showPreevyService || service !== COMPOSE_TUNNEL_AGENT_SERVICE_NAME)
       .map(tunnel => ({
         ...tunnel,
-        ports: mapValues(tunnel.ports, includeAccessCredentials ? addCredentials : (x: string) => x),
+        ports: mapValues(
+          tunnel.ports,
+          includeAccessCredentials
+            ? (x: string) => `${addCredentials(x)}?_preevy_auth_hint=basic`
```

Comment:
Adding the credentials using string concatenation might break if the URL already contains query parameters.

I suggest creating `setQueryParam` alongside `replaceHostname` at `common/src/urls.ts`